Userland/Libraries/LibWeb/HTML/HTMLStyleElement.cpp

```diff
@@ -138,6 +138,11 @@ void HTMLStyleElement::update_a_style_block()
     // FIXME: This should probably be handled by StyleSheet::set_owner_node().
     m_associated_css_style_sheet = sheet;
 
+    if (!attribute(HTML::AttributeNames::media).is_empty()) {
+        auto media_query_list = parse_media_query_list(CSS::Parser::ParsingContext(document()), attribute(HTML::AttributeNames::media));
+        sheet->set_media_query_list(media_query_list);
+    }
+
```

Comment:
We already pass the `media` attribute to `create_a_css_style_sheet()` below. I think `parse_media_query_list()` should happen there, in `StyleSheet::set_media()`.

Then the `MediaQueryList` should be exposed as `StyleSheet::media()`: we have it commented out [here](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/CSS/StyleSheet.idl#L15) but don't feel like you have to make that work if you don't want to! It's just, that makes it clear that `StyleSheet` is where this belongs. :^)

Sorry so much of this code is confusing, I forgot how much of a mess I left it in. :sweat_smile: